### PR TITLE
Fix/refactor admin posts

### LINKED ISSUE #73: Cannot post from Q&A and Matrix forums
Not sure when this was introduced, but even when logged in, I cannot post messages from these views.

### PR DESCRIPTION
**Pull request by scott**
This PR refactors the `_post` partial by moving to two partials one for admin and one for front-end.  This removed a lot of icky spagetti switching code and made it easier to resolve some issues related to trying to share one partial between two concerns.

I also merged in the integration testing branch and extended it, so thats included in this PR as well.

This PR fixes #73, #42 and #20

### CODE REVIEW

**Comment by scott** [comment]
@olliebennett can you do a quick review before I pull this?

File: app/views/layouts/_login.html.erb [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 <hr class="hidden-lg hidden-md"/>
 
-<div class="col-md-6">
+<div class="login-form" class="col-md-6">
```
Comment (olliebennett) [inline app/views/layouts/_login.html.erb]: Is this as reliable as `class="login-form col-md-6"`? No idea whether it's treated differently, but seems like the kind of thing to break in old browsers, for example. Can't find any info on what happens when an attribute is defined multiple times on an element like this.

**Comment by olliebennett** [comment]
Nice one :) This looks good - glad to see some integration tests too!

File: app/views/layouts/_login.html.erb (hunk 1 above)
Comment (scott) [inline app/views/layouts/_login.html.erb]: That looks like a mistake/typo good catch!

**Pull request merged**